File: desing-pattens-1/src/modelo/implementacao/imposto/ICMS.java
package modelo.implementacao.imposto;

import interfaces.IImposto;
import modelo.Orcamento;

public class ICMS implements IImposto {

	@Override
	public double calcula(Orcamento o) {

		return o.getValor() * 0.05 + 50;
	}

}

File: desing-pattens-1/src/main/java/modelo/implementacao/imposto/ICMS.kt
package main.java.modelo.implementacao.imposto

import main.java.interfaces.IImposto
import main.java.modelo.Orcamento

class ICMS : IImposto {

    override fun calcula(o: Orcamento): Double {

        return o.valor * 0.05 + 50
    }

}
